src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt

```diff
@@ -250,8 +249,27 @@ class NodeService(
           info =
             InfoDto().apply {
               type = authCompleted.transactionAuthorizationRequestData.paymentTypeCode
-              brandLogo = authCompleted.transactionAuthorizationRequestData.logo.toString()
-              brand = authCompleted.transactionAuthorizationRequestData.brand?.name
+              brandLogo =
+                authCompleted.transactionAuthorizationRequestData
+                  .transactionGatewayAuthorizationRequestedData
+                  .logo
+                  .toString()
+              brand =
+                when (authCompleted.transactionAuthorizationRequestData.paymentGateway) {
+                  TransactionAuthorizationRequestData.PaymentGateway.NPG -> {
+                    (authCompleted.transactionAuthorizationRequestData
+                        .transactionGatewayAuthorizationRequestedData
+                        as NpgTransactionGatewayAuthorizationRequestedData)
+                      .brand
+                  }
+                  else -> {
```

Comment:
What do you think to explicit the type of PaymentGateway and return a null value if nothing of the if branch match the kind of data. It is the same but I think that it could be clear for future evolution

Reply:
Refactored with https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/158/commits/2c340e23d4abd35a0105abfb5b0da5452ca24538